tests/test_hf_api.py

```diff
@@ -288,6 +288,76 @@ def test_move_repo_invalid_repo_id(self) -> None:
         with pytest.raises(ValueError, match=r"Invalid repo_id*"):
             self._api.move_repo(from_id="invalid_repo_id", to_id="namespace/repo_name")
 
+    def test_update_repo_settings(self):
+        repo_id = self._api.create_repo(repo_id=repo_name()).repo_id
+        # Test setting gated to "auto"
+        self._api.update_repo_settings(repo_id=repo_id, gated="auto")
+        res = self._api.model_info(repo_id, expand="gated")
+        assert res.gated == "auto"
+        # Test setting gated to "manual"
+        self._api.update_repo_settings(repo_id=repo_id, gated="manual")
+        res = self._api.model_info(repo_id, expand="gated")
+        assert res.gated == "manual"
+        # Test disabling gating
+        self._api.update_repo_settings(repo_id=repo_id, gated=False)
+        res = self._api.model_info(repo_id, expand="gated")
+        assert res.gated is False
+        self._api.delete_repo(repo_id=repo_id)
```

Comment:
```suggestion
    @use_tmp_repo(repo_type="model")
    def test_update_repo_settings(self, repo_url: RepoUrl):
        repo_id = repo_url.repo_id

        for gated_value in ["auto", "manual", False]:
            self._api.update_repo_settings(repo_id=repo_id, gated=gated_value)
            info = self._api.model_info(repo_id, expand="gated")
            assert info.gated == gated_value
```

The logic of the test can be made a bit more concise by:
- using `@use_tmp_repo` that will handle repo creation and deletion for you. With the current implementation if an error occurs, `self._api.delete_repo` is not called to clean up the test repo. The decorator handle both if the test succeeds or fails
- we can make the logic to test all 3 values a bit more compact given very little difference between each case. Writing tests is always a balance between simplicity and readability. In the case here, it's fine to add a for loop if it saves 6 duplicated lines of code

Reply:
That's a great suggestion! I appreciate your feedback on how to improve test logic.

Using the `@use_tmp_repo` decorator definitely makes the test code more concise and ensures proper cleanup even in case of errors.
Regarding the second point, I agree that we can compact the logic to test all three values of gated using a for loop. This would eliminate code duplication and enhance readability.

I'll keep these suggestions in mind when writing future tests. Thanks again for your input!